client/components/planning-editor-standalone/profile-converter.ts

```diff
@@ -0,0 +1,45 @@
+import {flatMap} from 'lodash';
+import {planningApi} from '../../superdeskApi';
+import {getEditorFormGroupsFromProfile} from '../../utils/contentProfiles';
+
+interface IBaseField<T> {
+    type: T;
+    fieldId: string;
+    required: boolean;
+}
+
+interface ICustomVocabularyField extends IBaseField<'custom_vocabulary'> {
+    vocabularyId: string;
+}
+
+type IFieldConverted = IBaseField<'normal'> | ICustomVocabularyField;
+
+export const getProfileFieldsConverted = (): Array<IFieldConverted> => {
```

Comment:
Avoid terminology(converting) if it's not helping significantly - because if you add terminology - you have to define it. Here I think it would be understandable enough to name it `getPlanningProfileFields`. You can add a comment about expansion of vocabulary fields.